tronweb2/src/components/NavBar/NavBar.js

```diff
@@ -1,10 +1,46 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import './NavBar.css';
 import {
   Link,
 } from 'react-router-dom';
+import Autosuggest from 'react-autosuggest';
+import { fetchFromApi } from '../../utils/utils';
+
+function jobLink(jobName) {
+  return <Link className="stretched-link" to={`/job/${jobName}`}>{jobName}</Link>;
+}
 
 function NavBar() {
+  const [inputSuggestions, setInputSuggestions] = useState([]);
+  const [inputValue, setInputValue] = useState('');
+  const [jobData, setJobData] = useState(undefined);
+  const maxSuggestions = 10;
+
+  useEffect(() => fetchFromApi('/api/jobs', setJobData), []);
+
+  let jobNames = [];
+  if (jobData !== undefined) {
+    if (!('error' in jobData)) {
+      jobNames = jobData.jobs.map((job) => job.name);
+    }
+  }
+
+  function getSuggestedJobNames(string) {
+    const trimmed = string.trim().toLowerCase();
+    const { length } = string;
+    return length === 0 ? [] : jobNames
+      .filter((jobName) => jobName.toLowerCase().slice(0, length) === trimmed)
+      .slice(0, maxSuggestions);
+  }
+
+  const inputProps = {
+    placeholder: 'Search jobs',
```

Comment:
I sorta wonder if there's value in having the search bar in both the navbar and the body of the "Scheduled Jobs" page, but either way.

Reply:
The Scheduled Jobs page has more info on it, so you can look at the state of several jobs by filtering, as opposed to the navbar for just navigating to a single job.